addons/digest/models/digest.py

```diff
@@ -164,6 +172,15 @@ def _cron_send_digest_email(self):
             except MailDeliveryException as e:
                 _logger.warning('MailDeliveryException while sending digest %d. Digest is now scheduled for next cron update.', digest.id)
 
+    def _get_unsubscribe_token(self, user_id):
+        """Generate a secure hash for this digest and user. It allows to
+        unsubscribe from a digest while keeping some security in that process.
+
+        :param int user_id: ID of the user to unsubscribe
+        """
+        self.ensure_one()
```

Comment:
Useless `ensure_one()`, if you call it with multiple records, you already get an `Expected singleton` error the next line.